envi/memory.py

```diff
@@ -525,31 +525,78 @@ def getMemoryMap(self, va):
     def getMemoryMaps(self):
         return [mmap for mva, mmaxva, mmap, mbytes in self._map_defs]
 
-    def readMemory(self, va, size):
+    def readMemory(self, va, size, origva=None):
         '''
         Read memory from maps stored in memory maps.
+
+        If the read crosses memory maps and fails on a later map, the exception
+        will show the details of the last map/failure, but should include the
+        original va (not the size).
         '''
         for mva, mmaxva, mmap, mbytes in self._map_defs:
             if mva <= va < mmaxva:
                 mva, msize, mperms, mfname = mmap
                 if not mperms & MM_READ:
-                    raise envi.SegmentationViolation(va)
+                    msg = "Bad Memory Read (no READ permission): %s: %s" % (hex(va), hex(size))
+                    if origva:
+                        msg += " (original va: %s)" % hex(origva)
+                    raise envi.SegmentationViolation(va, msg)
```

Comment:
in a best case scenario, i'd like to see the va/size/orig va as discrete fields that can be extracted programmatically. the freeform text here works for human consumption but not for exception handler logic.

but, since i don't have a concrete use right now, i can open a PR with the enhancement (should be strictly an addition, no breaking change) when i need it.

Reply:
this is lower priority, more just commentary.

Reply:
thanks.  will look forward to it.